src/Sylius/Bundle/ResourceBundle/Form/EventSubscriber/AddCodeFormSubscriber.php

```diff
@@ -38,14 +39,20 @@ public static function getSubscribedEvents()
     public function preSetData(FormEvent $event)
     {
         $resource = $event->getData();
+        $type = 'text';
 
         if (!$resource instanceof CodeAwareInterface) {
             throw new UnexpectedTypeException($resource, CodeAwareInterface::class);
         }
 
+        if($resource instanceof CurrencyInterface){
```

Comment:
By the way, it would be nice to have this subscriber only modifying existing `code` type. Currently you can't pass your own options here.